library/ssl_msg.c

```diff
@@ -4798,6 +4798,9 @@ size_t mbedtls_ssl_get_bytes_avail( const mbedtls_ssl_context *ssl )
 
 int mbedtls_ssl_check_pending( const mbedtls_ssl_context *ssl )
 {
+    if( ssl == NULL || ssl->conf == NULL )
+        return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
```

Comment:
This function returns 0 (no pending input) or 1 (pending input), not an error.

Since no input can be read on an invalid context, I think it makes sense to return 0. Like Andrzej I think this should be documented.